### PR TITLE
Change position and enhance top bar elements

### PR DESCRIPTION
**Pull request by lordelix**
## Description

Reposition Elements appearing in the top bar. Replace the left caret with a burger menu.

## Motivation and Context

Requirement from by Marketing and CEOs

## How Has This Been Tested?

Visual test done on the latest Versions of:
- Google Chrome
- Firefox
- MS Edge

## Screenshots:

ec_reworking_header

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/css/header.css [hunk 1]
```diff
@@ -292,28 +321,52 @@
...
+}
+
 #expand:hover,
 #expand:focus,
 #expand:active {
 	color: #fff;
 }
+
+#expand:hover:after {
+	border-color: #fff transparent transparent transparent;
```
Comment (phisch) [inline core/css/header.css]: much transparent :)
how about `border-top-color`? :P

File: core/css/header.css [hunk 2]
```diff
@@ -82,65 +92,84 @@
...
+	padding: 0;
+	padding-left: 66px;
+	font-size: 16px;
+	font-weight: 300;
+	line-height: inherit;
 }
 
 #header .header-appname-container {
-	display: inline-block;
+	display: flex;
```
Comment (phisch) [inline core/css/header.css]: Same here, very poor support in our officially supported browsers. But looks like this should be fine if you add `-moz-` and `-webkit-` prefixes.

File: core/css/header.css [hunk 3]
```diff
@@ -49,24 +49,34 @@
 #owncloud {
 	position: absolute;
 	top: 0;
-	left: 0;
+	left: calc(50% - 75px);
```
Comment (phisch) [inline core/css/header.css]: Officially we support IE 11+, FF 14+, Chrome 18+ and Safari 5+
`calc` needs the `-moz-` prefix for FF 14 and 15, is not supported in Chrome 18, needs the `-webkit-` prefix for Chrome 19-25, is not supported in Safari 5 and 5.1, and also needs the `-webkit-` prefix for Safari 6.

I would not mind bumping the required browser versions. Maybe even better would be an acceptable workaround for those.

What do you think @felixheidecke?

**Comment by phil-davis** [comment]
I raised issue #28016 to review supported browser versions, because it would be nice, if reasonably possible, to not have to think too hard about old browser versions when making new code/changes.

File: core/css/header.css (hunk 3 above)
Comment (felixheidecke) [inline core/css/header.css]: Bumping up the versions is a must: Here is my suggestion ;-)
https://github.com/owncloud/core/issues/28016#issuecomment-304610698

File: core/css/header.css (hunk 1 above)
Comment (felixheidecke) [inline core/css/header.css]: I had to reset all other colors too.

**Comment by felixheidecke** [comment]
@SergioBertolinSG I changed the template in the files_sharing wich, in turn, will [fix this bug](https://github.com/owncloud/qa-enterprise/issues/87#issuecomment-307380888) in the EE Theme as well.

Latest changes: https://github.com/owncloud/core/pull/27986/commits/39046776747705840edf6ee64e9644257f356f7c

**Comment by jvillafanez** [comment]
Not sure what happened here but this is a big nope 🚫 

@felixheidecke please, pick your changes and move them to a new fresh branch from master, and close this PR. (Sorry if this sound a bit hard, but I don't know who's going to review 102 commits or  283 different files)

**Comment by felixheidecke** [comment]
Please see new PR here: https://github.com/owncloud/core/pull/28287

**Pull request closed without merging**